### PR TITLE
NIFI-8957 NiFi Registry - Possibility to set a description when creating a bucket

### PR DESCRIPTION
**Pull request by adenes**
- added "description" column to the bucket list table
- added "description" field to the "new bucket" popup
- added "description" field to the bucket editing sidebar
- updated the UI tests

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/dialogs/create-bucket/nf-registry-create-bucket.js [hunk 1]
```diff
@@ -80,21 +81,22 @@ NfRegistryCreateBucket.prototype = {
...
-    },
-
-    /**
-     * Focus the new bucket input.
-     */
-    ngAfterViewChecked: function () {
-        this.newBucketInput.nativeElement.focus();
     }
+
+//    /**
```
Comment (markobean) [inline nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/dialogs/create-bucket/nf-registry-create-bucket.js:86]: Should this section be removed? (Was it left as comment accidentally?)

File: nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js [hunk 2]
```diff
@@ -462,6 +468,7 @@ NfRegistryManageBucket.prototype = {
                             if (!response.status || response.status === 200) {
                                 self.nfRegistryService.bucket = response;
                                 self.bucketname = self.nfRegistryService.bucket.name;
+                                self.description = self.nfRegistryService.bucket.description;
```
Comment (markobean) [inline nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/sidenav/manage-bucket/nf-registry-manage-bucket.js:471]: There are similar code patters to this one at/about lines 530 and 589. Does "self.description" need to be added in these places as well?

**Comment by markobean** [comment]
Built and installed the nifi-registry with this addition. The UI appeared to perform as expect. The Description field of each bucket is editable. I started from an empty Registry, and have not tested if there are any issues when upgrading from an existing Registry.

Connected this registry to a running NiFi. The bucket description information is available when hovering over the "?" next to the bucket name when saving or importing a flow.

Functionally, this looks good. Would appreciate someone with more JavaScript experience to do another look-over of the code though.

File: nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/dialogs/create-bucket/nf-registry-create-bucket.js (hunk 1 above)
Comment (adenes) [reply nifi-registry/nifi-registry-core/nifi-registry-web-ui/src/main/webapp/components/administration/workflow/dialogs/create-bucket/nf-registry-create-bucket.js:86]: That should be removed indeed, thanks for pointing it out. I'll update the PR accordingly.

**Comment by adenes** [comment]
Thanks @markobean for the review. I have updated the PR according to your suggestions.

**Review by mtien-apache — APPROVED** [review]
Nice work! Verified a successful full build, a successful frontend build by running `build:webapp:bundle:production`, and I'm able to set and edit the Bucket description. 

I only have one minor comment, otherwise all looks good. Thanks!

**Review by pvillard31 — APPROVED** [review]

**Comment by pvillard31** [comment]
Merged, thanks @adenes @markobean @mtien-apache

**Pull request closed without merging**